src/MFPEncoder.ts

```diff
@@ -105,83 +111,114 @@ export default class MFPBaseEncoder {
   /**
    * Encodes a given OpenLayers layerState to Mapfish print format.
    * @param layerState
-   * @param printResolution
    * @param customizer
    * @return a spec fragment
    */
   async encodeLayerState(
     layerState: State,
-    printResolution: number,
     customizer: BaseCustomizer,
   ): Promise<MFPLayer[] | MFPLayer | null> {
     if (
       !layerState.visible ||
-      printResolution < layerState.minResolution ||
-      printResolution >= layerState.maxResolution
+      this.printResolution < layerState.minResolution ||
+      this.printResolution >= layerState.maxResolution
     ) {
       return null;
     }
     const layer = layerState.layer;
 
-    if (layer instanceof VectorTileLayer) {
-      return await this.encodeMVTLayerState(layerState, printResolution, customizer);
+    if (layer instanceof ImageLayer) {
+      return this.encodeImageLayerState(layerState, customizer);
+    }
+
+    if (layer instanceof VectorLayer) {
+      const encoded = new VectorEncoder(layerState, customizer).encodeVectorLayer(this.printResolution)!;
+      this.addRenderAsSVG(layerState, encoded);
+      return encoded;
     }
 
     if (layer instanceof TileLayer) {
       return this.encodeTileLayerState(layerState, customizer);
-    } else if (layer instanceof VectorLayer) {
-      const encoded = new VectorEncoder(layerState, customizer).encodeVectorLayer(printResolution)!;
-      const renderAsSvg = layer.get('renderAsSvg');
-      if (renderAsSvg !== undefined) {
-        encoded.renderAsSvg = renderAsSvg;
-      }
-      return encoded;
+    }
+
+    if (layer instanceof VectorTileLayer) {
+      return await this.encodeMVTLayerState(layerState, customizer);
     }
 
     return null;
   }
 
   /**
-   *
-   * @param layerState An MVT layer state
-   * @param printResolution
-   * @param customizer
-   * @return a spec fragment
+   * Get "renderAsSvg" to the encoded object if it exists in the layer.
```

Comment:
```suggestion
   * Set "renderAsSvg" to the encoded object if it exists in the layer.
```

Reply:
Should it be private?

Reply:
protected maybe

Reply:
In fact, having this private SCHM parameter here is discutable. We should remove it.
And in all cases, I would not make this method private or protected because no other method are private/protected

Reply:
That's like every `layer.get('name')`, a project can have the "name" of the layer in a "title" property, that's not generic and should be possible to override.

Reply:
`renderAsSvg` is a MFP feature. It is not specific to Schweizmobil.
It could have a better name though: for example `mfpRenderAsSvg`.
The reason it should be private is that there is no reason to change it.

Regarding the `name`, yes it could be made configurable but I don't know if it is worth it. According to @sbrunner this is not really used in MFP. In Schweizmobil I used it to help debugging.

Reply:
But I was speaking about the `name` and `renderAsSvg` properties in the ol layer. That's not standard. That's a property and not every project will follow this naming.
**But that's not part of the aim of this PR, I've reverted this change.**